src/P/Function.hs

```diff
@@ -15,6 +22,20 @@ import           Data.List (foldr, replicate)
 import           Data.Function (id, const, (.), flip, ($))
 import           Data.Function (fix, on)
 
+#if (__GLASGOW_HASKELL__ >= 710)
+
+import           Data.Function ((&))
+
+#else
+
+-- | '&' is a reverse application operator.  This provides notational
+--   convenience.  Its precedence is one higher than that of the forward
+--   application operator '$', which allows '&' to be nested in '$'.
```

Comment:
Wow looks like they define a precedence for infix `on`, didn't know that was possible